src/compile/scale/properties.ts

```diff
@@ -271,6 +272,11 @@ export function padding(
       const {type, orient} = markDef;
       if (type === 'bar' && !(isFieldDef(fieldOrDatumDef) && (fieldOrDatumDef.bin || fieldOrDatumDef.timeUnit))) {
         if ((orient === 'vertical' && channel === 'x') || (orient === 'horizontal' && channel === 'y')) {
+          const sizeChannel = getSizeChannel(channel);
```

Comment:
This is not exactly offset, but it's a relevant bug that bar should respect relative band size.